projects/kit/components/input-date-multi/input-date-multi.component.ts

```diff
@@ -132,16 +133,24 @@ export class TuiInputDateMultiComponent
 
     protected open = false;
 
-    protected readonly dateFormat = inject(TUI_DATE_FORMAT);
-    protected readonly dateSeparator = inject(TUI_DATE_SEPARATOR);
+    protected dateFormat = TUI_DEFAULT_DATE_FORMAT;
     protected readonly isMobile = inject(TUI_IS_MOBILE);
     protected readonly doneWord$ = inject(TUI_DONE_WORD);
     protected readonly filler$: Observable<string> = this.dateTexts$.pipe(
         map(dateTexts =>
-            changeDateSeparator(dateTexts[this.dateFormat], this.dateSeparator),
+            changeDateSeparator(
+                dateTexts[this.dateFormat.mode],
+                this.dateFormat.separator,
+            ),
         ),
     );
 
+    protected readonly dateFormat$ = inject(TUI_DATE_FORMAT)
+        .pipe(tuiWatch(this.cdr), takeUntil(this.destroy$))
```

Comment:
Let's migrate to `takeUntilDestroyed` while we're at it here and in other places where you use destroy service.